cmd/kubeadm/app/cmd/phases/reset/removeetcdmember.go

```diff
@@ -68,7 +71,14 @@ func runRemoveETCDMemberPhase(c workflow.RunData) error {
 				fmt.Println("[reset] Would remove the etcd member on this node from the etcd cluster")
 			}
 		}
+		if err := CleanDir(etcdDataDir); err != nil {
+			klog.Warningf("[reset] Failed to delete contents of %q directory: %v", etcdDataDir, err)
+		}
 	} else {
+		// In case `cleanup-node` phase is executed before the `remove-etcd-member` phase, kubeadm will perform best effort cleanup.
+		// Only the default etcd data dir will be cleanup which means if the etcd data is set to the value other than
+		// the default value, you might want to manually cleanup the etcd data.
+		klog.Warning("[reset] Manually cleanup etcd data might be needed if the cluster is reset by each phase")
```

Comment:
thinking about this more the behavior is confusing and this message is too.
we should at least print the location that the user must cleanup manually.

i think our general; goal should be to make reset idempotent, if a phase is called and then another phase or the whole command is called kubeadm should not error and just perform cleanups even if two phases have to do the same cleanup...not sure what is best here.

Reply:
> i think our general goal should be to make reset idempotent
Agree, this is what this pr try to achieve.

> we should at least print the location that the user must cleanup manually.
This is hard to tell since the config file has been removed by the `cleanup-node`, anywhere else we can get it? If we know where is the data persisted, we can just remove instead of asking for manually removal.

Reply:
> klog.Warning("[reset] Manually cleanup etcd data might be needed if the cluster is reset by each phase")

now that we cannot tell where is the etcd data is configured, I feel we can just remove this line.